src/librustc_data_structures/small_vec.rs

```diff
@@ -209,11 +209,7 @@ impl<A> Decodable for SmallVec<A>
           A::Element: Decodable {
     fn decode<D: Decoder>(d: &mut D) -> Result<SmallVec<A>, D::Error> {
         d.read_seq(|d, len| {
-            let mut vec = SmallVec::with_capacity(len);
```

Comment:
I have a hunch that this was deliberately done for optimization

cc @nnethercote

Reply:
I don't have a good sense of how `collect()` works. Will it use `size_hint` from the range to pre-allocate an appropriate capacity?

Reply:
It should, yes.